R/GMA.R

```diff
@@ -0,0 +1,445 @@
+                            cores = cores)
+  
+  # Convert functional pairs results from EpiMix to tibble
+  EpiMix_regular_FP <- as_tibble(EpiMix_regular$FunctionalPairs) %>% 
+    dplyr::rename("Hugo_Symbol" = "Gene")
+  EpiMix_enhancer_FP <- as_tibble(EpiMix_enhancer$FunctionalPairs) %>% 
+    dplyr::rename("Hugo_Symbol" = "Gene")
+  
+  # Add enhancer prefix to columns in EpiMix_enhancer_FP
+  # Create column that specifies that identified probes are in enhancers 
+  EpiMix_enhancer_FP <- EpiMix_enhancer_FP %>% 
+    dplyr::rename_with(.cols = -1, 
+                       .fn = ~str_c(., "_enhancer")) %>% 
+    dplyr::mutate(Probe_in_enhancer = "Enhancer") %>% 
+    dplyr::relocate(Probe_in_enhancer, 
+                    .after = Probe_enhancer) 
+  
+  
+  ## Initialize data --------------------------------------------------------
+  
+  # Convert PRA data to tibble
+  PRA_tbl <- PRAtoTibble(dataPRA = dataPRA)
+  
+  # Get NCG data
+  NCG <- get("NCG")
+  
+  
+  ## DEG_Methylation_Annotations --------------------------------------------
+  
+  ## Intersect DEA results with EpiMix results
+  ## Create DEG_Methylation_Annotations
+  
+  # Convert DEA data to tibble
+  dataDEGs <- dataDEGs %>% 
+    tibble::rownames_to_column(var = "Hugo_Symbol") %>% 
+    as_tibble()
+  
+  # Bind EpiMix regular and enhancer functional-pairs results
+  # Join with DEA data and with PRA data containing information 
+  # about oncogenic mediators
+  # Add NCG annotations
+  DEG_Methylation_Annotations <- EpiMix_regular_FP %>% 
+    bind_rows(EpiMix_enhancer_FP) %>% 
+    right_join(x = .,
+               y = dataDEGs,
+               by = "Hugo_Symbol") %>% 
+    left_join(x = .,
+              y = PRA_tbl,
+              by = "Hugo_Symbol") %>%
+    left_join(x = .,
+              y = NCG,
+              by = c("Hugo_Symbol" = "symbol")) %>%
+    dplyr::select(-c("Fold change of gene expression",
+                     "Fold change of gene expression_enhancer"))
+  
+  ## Get chromosome, start and end locations of probes
+  
+  # Retrieve annotations of probes and select only chromosome, start and
+  # end positions
+  probe_annotation <- EpiMix_getInfiniumAnnotation(plat = met_platform, 
+                                                   genome = "hg38") %>% 
+    BiocGenerics::as.data.frame() %>% 
+    as_tibble(rownames = "Probe") %>% 
+    dplyr::select(Probe,
+                  seqnames, 
+                  start,
+                  end) %>% 
+    dplyr::rename(Chromosome = seqnames)
+  
+  # Add probe annotations (chromosome, start and end positions) 
+  # to DEG_Methylation_Annotations 
+  DEG_Methylation_Annotations <- DEG_Methylation_Annotations %>% 
+    left_join(x = .,
+              y = probe_annotation,
+              by = "Probe") %>% 
+    left_join(x = .,
+              y = probe_annotation,
+              by = c("Probe_enhancer" = "Probe")) %>% 
+    unite(start, 
+          start.x, 
+          start.y, 
+          na.rm = TRUE) %>% 
```

Comment:
It is because I'm uniting two columns, called `start.x` and `start.y`, into one column called `start`. Some of the values in `start.x` and `start.y` are NAs, e.g. `start.x = 51908186` and `start.y = NA` and then if we use `na.rm = FALSE`, then `start` will be `51908186_NA`, but if we use `na.rm = TRUE`, then we get `start = 51908186` which is what we want. Then afterwards, I'm converting `start` to numeric